data-processing/entities/definitions/nlp.py

```diff
@@ -309,18 +305,16 @@ def featurize(self, text: str, limit: bool = False) -> DefaultDict[Any, Any]:
         return features
 
     def predict_batch(
-        self, data: List[Dict[Any, Any]], prediction_type: str
+        self, data: List[Dict[Any, Any]]
     ) -> Tuple[Dict[Any, Any], Dict[str, List[List[str]]], Dict[Any, Any]]:
 
         # Load data.
         test_dataset, raw = load_and_cache_example_batch_raw(
-            self.data_args[prediction_type], self.tokenizer[prediction_type], data,
+            self.data_args, self.tokenizer, data,
         )
 
         # Perform inference.
-        intent_pred, slot_preds, slot_pred_confs = self.trainer[
-            prediction_type
-        ].evaluate_from_input(test_dataset, raw)
+        intent_pred, slot_preds, slot_pred_confs = self.trainer.evaluate_from_input(test_dataset, raw)
 
```

Comment:
@andrewhead Also while predicting, we call the args, tokenizer and model directly as opposed to accessing it via a dictionary element.

Reply:
great!